cmd/redshift.go

```diff
@@ -70,34 +71,45 @@ func NewRedshiftCmd() *cobra.Command {
 		if err != nil {
 			return errors.Trace(err)
 		}
-		_, sourceTable := utils.SplitTableFQN(tableFQN)
-		db, err := redshiftConfigFromCli.OpenDB()
-		if err != nil {
-			return errors.Trace(err)
-		}
-		snapConnector, err := redshiftsql.NewRedshiftConnector(
-			db,
-			redshiftConfigFromCli.Schema,
-			fmt.Sprintf("snapshot_external_%s", sourceTable),
-			redshiftConfigFromCli.Role,
-			snapshotURI,
-			credValue,
-		)
-		if err != nil {
-			return errors.Trace(err)
-		}
-		increConnector, err := redshiftsql.NewRedshiftConnector(
-			db,
-			redshiftConfigFromCli.Schema,
-			fmt.Sprintf("increment_external_%s", sourceTable),
-			redshiftConfigFromCli.Role,
-			incrementURI,
-			credValue,
-		)
-		if err != nil {
-			return errors.Trace(err)
+
+		snapConnectorMap := make(map[string]coreinterfaces.Connector)
+		increConnectorMap := make(map[string]coreinterfaces.Connector)
+		for _, tableFQN := range tables {
+			_, sourceTable := utils.SplitTableFQN(tableFQN)
```

Comment:
One table FQN failure will stop all?

Reply:
if one table open connector fails, all tables should be failed.